lib/rules/no-object-constructor.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Rule to disallow calls to the `Object` constructor without an argument
+ * @author Francesco Trotta
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const { getVariableByName, isArrowToken } = require("./utils/ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "Disallow calls to the `Object` constructor without an argument",
+            recommended: false,
+            url: "https://eslint.org/docs/latest/rules/no-object-constructor"
+        },
+
+        hasSuggestions: true,
+
+        schema: [],
+
+        messages: {
+            preferLiteral: "The object literal notation {} is preferable.",
+            useLiteral: "Replace with '{{replacement}}'."
+        }
+    },
+
+    create(context) {
+
+        const sourceCode = context.sourceCode;
+
+        /**
+         * Determines whether or not an object literal that replaces a specified node needs to be enclosed in parentheses.
+         * @param {ASTNode} node The node to be replaced.
+         * @returns {boolean} Whether or not parentheses around the object literal are required.
+         */
+        function needsParentheses(node) {
+            let exprStatNode = node;
+
+            while (exprStatNode && exprStatNode.type !== "ExpressionStatement") {
+                exprStatNode = exprStatNode.parent;
+            }
```

Comment:
Maybe this loop could end earlier if `exprStatNode.range[0]` becomes `< node.range[0]`?

Reply:
Thanks! I've moved to logic to a separated function, this should also help to clarify the intent: https://github.com/eslint/eslint/pull/17576/commits/97dd022050493f1bf6c9aa37042024d3698e568e.